### PR TITLE
Add azure_route support routeTableResourceGroup

### PR DESCRIPTION
**Pull request by suker200**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**:
This PR adds a new parameter in the Azure Cloud Provider config file.
The new parameter is RouteTableResourceGroup. It has the same behavior as the VNetResourceGroup has for Vnet. Because sometimes, you want to use an Azure RouteTable which is not is in the same Resource Group as your main Resource Group

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
Azure Cloud Provider: Add new parameter `RouteTableResourceGroup`

```release-note
Support specify the Resource Group of Route Table when update Pod network route (Azure)
```

This pr just origin from https://github.com/kubernetes/kubernetes/pull/71321 . Thanks titilambert

### CODE REVIEW

**Comment by suker200** [comment]
I signed it

**Comment by suker200** [comment]
@justaugustus  would you please help me review on this

**Review by yastij — APPROVED** [review]
/lgtm
/priority backlog

File: pkg/cloudprovider/providers/azure/azure_test.go [hunk 1]
```diff
@@ -941,6 +943,7 @@ func getTestCloud() (az *Cloud) {
 			},
 			ResourceGroup:                "rg",
 			VnetResourceGroup:            "rg",
+			RouteTableResourceGroup:      "rg",
```
Comment (justaugustus) [inline pkg/cloudprovider/providers/azure/azure_test.go]: Should these have different names, e.g., `rtb-rg` and `vnet-rg`?

**Comment by justaugustus** [comment]
LGTM, but I'd Pengfei or Andy to review as well.
/assign @feiskyer @andyzhangx

File: pkg/cloudprovider/providers/azure/azure_test.go [hunk 2]
```diff
@@ -1558,6 +1562,8 @@ aadClientSecret: --aad-client-secret--
 }
 
 // Test Configuration deserialization (yaml)
+// Test Configuration deserialization (yaml) without
```
Comment (feiskyer) [inline pkg/cloudprovider/providers/azure/azure_test.go]: nit: `Test Configuration deserialization (yaml)` dup with above line

**Comment by feiskyer** [comment]
@suker200 Could you address the comments and add a release note?

File: pkg/cloudprovider/providers/azure/azure_test.go (hunk 2 above)
Comment (suker200) [reply pkg/cloudprovider/providers/azure/azure_test.go]: done

**Comment by suker200** [comment]
@feiskyer Done. Thank you

**Review by feiskyer — COMMENTED** [review]
Looks good. Could you squash the commits?

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**